src/System.Windows.Forms/src/System/Windows/Forms/ListView.ListViewAccessibleObject.cs

```diff
@@ -0,0 +1,240 @@
+                var selectedItems = OwningListView.SelectedItems;
+                foreach (var selectedItem in selectedItems)
+                {
+                    selectedItemProviders.Add(
+                        new ListViewItemAccessibleObject(OwningListView, selectedItem));
+                }
+
+                return selectedItemProviders.ToArray();
+            }
+
+            internal override UiaCore.IRawElementProviderFragment ElementProviderFromPoint(double x, double y)
+            {
+                return HitTest((int)x, (int)y);
+            }
+
+            public override AccessibleObject HitTest(int x, int y)
+            {
+                Point point = OwningListView.PointToClient(new Point(x, y));
+                ListViewHitTestInfo hitTestInfo = OwningListView.HitTest(point.X, point.Y);
+                if (hitTestInfo.Item != null)
+                {
+                    if (hitTestInfo.SubItem != null)
+                    {
+                        return new ListViewSubItemAccessibleObject(OwningListView, hitTestInfo.Item, hitTestInfo.SubItem);
+                    }
+
+                    return new ListViewItemAccessibleObject(OwningListView, hitTestInfo.Item);
+                }
+
+                return null;
+            }
+
+            internal override int ColumnCount
+            {
+                get
+                {
+                    int columnCount = 0;
+
+                    var items = OwningListView.Items;
+                    if (items.Count > 0)
+                    {
+                        columnCount = items[0].SubItems.Count;
+                    }
+
+                    return columnCount;
+                }
+            }
+
+            internal override int RowCount => OwningListView.Items.Count;
+
+            internal override UiaCore.IRawElementProviderSimple[] GetRowHeaders() => null;
+
+            internal override UiaCore.IRawElementProviderSimple[] GetColumnHeaders()
+            {
+                var columnHeaders = new List<UiaCore.IRawElementProviderSimple>();
+                foreach (var columnHeader in OwningListView.Columns)
+                {
+                    columnHeaders.Add(new ListViewColumnAccessibleObject(columnHeader as ColumnHeader));
+                }
+
+                return columnHeaders.ToArray();
+            }
+
+            internal override UiaCore.RowOrColumnMajor RowOrColumnMajor => UiaCore.RowOrColumnMajor.RowMajor;
+
+            internal override int GetMultiViewProviderCurrentView()
+            {
+                return (int)OwningListView.View;
+            }
+
+            internal override int[] GetMultiViewProviderSupportedViews()
+            {
+                var allViews = Enum.GetValues(typeof(View));
+                var allViewIds = new List<int>();
+                foreach (var view in allViews)
+                {
+                    allViewIds.Add((int)view);
+                }
+
+                return allViewIds.ToArray();
```

Comment:
Why not perform a direct cast?
```suggestion
                return values as int[];
```

It will be faster and requires 64% less memory:

|                Method |       Mean |    Error |   StdDev |     Median | Ratio | RatioSD | Rank |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------------------- |-----------:|---------:|---------:|-----------:|------:|--------:|-----:|-------:|------:|------:|----------:|
| Allocate_List_ToArray | 1,132.7 ns | 22.56 ns | 39.52 ns | 1,115.0 ns |  1.00 |    0.00 |    3 | 0.0591 |     - |     - |     376 B |
|        Allocate_array | 1,029.2 ns | 19.47 ns | 19.12 ns | 1,022.0 ns |  0.91 |    0.04 |    2 | 0.0420 |     - |     - |     272 B |
|            Cast_array |   713.4 ns |  7.56 ns |  6.70 ns |   713.6 ns |  0.62 |    0.02 |    1 | 0.0210 |     - |     - |     136 B |

Reply:
How often do we expect these methods to run? May be it is work caching the `int[]` array instead of constantly reflecting over the enum?

Reply:
Yes, thanks, moved to read-only fields.